packages/dom/src/phrasing-content.js

```diff
@@ -47,8 +56,33 @@ const textContentSchema = {
 	bdo: { attributes: [ 'dir' ] },
 	wbr: {},
 	'#text': {},
+	...specialTags,
 };
 
+export const alterSpecialTags = ( HTML ) =>
+	Object.entries( specialTags ).reduce( ( acc, [ tag, { originalTag } ] ) => {
+		const openingTagRegExp = new RegExp( `<${ tag }`, 'g' );
+		const closingTagRegExp = new RegExp( `</${ tag }`, 'g' );
+
+		const withOriginalOpeningTag = HTML.replace(
```

Comment:
I would walk the tree instead of relying on regexps - they may have unintended consequences.

Reply:
All manually placed special chars like < have escape characters so this should not be dangerous, nevertheless, indeed tree traversing is just safe, so it is better to approach, thanks.
I am just not convinced with the entire idea - as I've mentioned in the desc - this is not scalable and there would be needed a huge amount of logic to cover all cases (colors are just a tip of an iceberg). I think it will be better to change the mechanics of unfolding letters from their corresponding spans, but I need feedback on this.